File: src/main/java/rusk/rest/TaskListResource.java
package rusk.rest;

import javax.inject.Inject;
import javax.ws.rs.GET;
import javax.ws.rs.Path;

import rusk.domain.list.TaskList;
import rusk.service.InquireTaskListService;

@Path("task-list")
public class TaskListResource {
    
    private InquireTaskListService service;
    
    @Inject
    public TaskListResource(InquireTaskListService service) {
        this.service = service;
    }
    
    @GET
    public TaskList inquire() {
        System.out.println("inquire task list");
        return null;
    }
}

File: src/main/ts/rusk/resource/list/TaskListResource.ts
module rusk {
    export module resource {
        export module list {
            export class TaskListResource {
                private $http : ng.IHttpService;
                
                constructor($http : ng.IHttpService) {
                    this.$http = $http;
                }
                
                inquire(callback : (taskList : rusk.model.list.TaskList) => void) : void {
                    this.$http.get('rest/task-list', {})
                        .success((data : rusk.model.list.TaskList) => {
                            callback(data);
                        });
                }
            }
        }
    }
}

angular
.module('rusk')
.factory('taskListResource', [
    '$http',
    ($http) => {
        return new rusk.resource.list.TaskListResource($http);
    }
]);
